6.traits.md

```diff
@@ -62,26 +62,29 @@ The top-level attributes of a trait define its metadata, version, kind, and spec
 | `metadata` | [`Metadata`](2.overview_and_terminology.md#metadata) | Y | | Trait metadata. |
 | `spec`| [`Spec`](#spec) | Y || A specification for trait attributes. |
 
 ### Spec
 
-The specification defines two major things: The list of workload types to which this trait applies, and the list of configurable parameters on this trait.
+The specification defines two major things: The list of workloads to which this trait applies, and the definition reference which will define the schema of this trait.
 
 | Attribute | Type | Required | Default Value | Description |
 |-----------|------|----------|---------------|-------------|
-| `appliesTo` | `[]string` | Y | `["*"]` | The list of workload types that this trait applies to. `"*"` means _any workload type_. A trait must apply to at least one workload type. This attribute must contain at least one value. An empty array is invalid for this attribute. |
-| `properties` | [`Properties`](#properties) | Y | | The trait's configuration options. The value is a [JSON schema](https://json-schema.org/) expressed as json string |
+| `appliesToWorkloads` | `[]string` | Y | `["*"]` | The list of workloads that this trait applies to. `"*"` means _any workload type_. A trait must apply to at least one workload type. This attribute must contain at least one value. An empty array is invalid for this attribute. |
+| `definitionRef` | [`DefinitionRef`](#definitionRef) | Y | | Index of the trait definition. |
 
-### Properties
+### DefinitionRef
 
-A properties object is an object whose structure is determined by the trait property schema. It may be a simple value, or it may be a complex object.
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `name` | `string` | Y | | The name of this trait definition. |
+| `version` | `string` | Y | | The version of this trait definition. |
+| `schematic` | `string` | N | | The schematic of this trait properties. The value is a [JSON schema](https://json-schema.org/). |
 
-The value of the properties field in a trait definition is a [JSON schema](https://json-schema.org/) expressed as json string. When a trait is applied to a component instance, the properties are validated against the schema defined in the trait's `properties` attribute.
+A trait definitionRef __MUST__ must be unique by `name` and `version`. The `schematic` field is optional as the definitionRef could be simply an indexing for a schema. In this case a runtime itself __SHOULD__ store definitions and a definition __MUST__ contain a schematic which can validate trait properties in application configurations.
 
-For example, if a trait defines a schema for properties that requires an array of integers with at least one member, the properties object is expected to be an array of integers with at least one member. During validation of the properties object, the trait's property schema will be applied.
 
 ## Core traits
```

Comment:
similarly, can we move the core traits definition out to a different place?

Reply:
good idea.